WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/CouponInputTransformer.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+import Yosemite
+
+/// Helper to update an `order` given an `OrderCouponLine` input type.
+///
+struct CouponInputTransformer {
+
+    /// Adds, deletes, or updates a coupon line input into an existing order.
+    ///
+    static func update(input: OrderCouponLine?, on order: Order) -> Order {
+        // If input is `nil`, then we remove the first existing coupon line.
+        guard let input = input else {
+            let updatedLines = order.coupons.enumerated().map { index, line -> OrderCouponLine? in
+                if index == 0 {
+                    return nil
+                }
+                return OrderFactory.newOrderCouponLine(code: line.code)
+            }.compactMap({ $0 })
+            return order.copy(coupons: updatedLines)
+        }
+
+        // If there is no existing coupon lines, we insert the input one.
+        guard order.coupons.count > 0 else {
```

Comment:
Nit: let's check for not empty

```suggestion
        guard order.coupons.isNotEmpty else {
```

Reply:
Done in 86db93f3abf2b08895ee51b943afedb1647f80c6